packages/simple-schema/lib/content-assist/element-name.js

```diff
@@ -51,7 +44,22 @@ function elementNameCompletion(elementNode, xssElement, prefix = "") {
   });
 
   if (namespacePrefix === undefined) {
-    const namespaces = filter(elementNode.namespaces, _ => !!_.prefix);
+    const namespaces = filter(elementNode.namespaces, namespace => {
+      const possibleElements = filter(
+        xssElement.elements,
+        element =>
+          has(element, "namespace") === true &&
+          element.namespace === namespace.uri
+      );
+      const possibleSuggestionsWithoutExistingSingular = applicableElements(
+        xssElement.elements,
+        elementNode.subElements,
+        possibleElements
+      );
+      const namespaceHasApplicableElements =
+        possibleSuggestionsWithoutExistingSingular.length > 0;
+      return !!namespace.prefix && namespaceHasApplicableElements;

```

Comment:
You are combining somewhat separate conditions here.
Maybe make a separate filtering (BEFORE) for ignoring the default namespace.

BTW I am going to change how namespaces are represented:
- clearer default namespace
- allow namespaces to be "deleted" according to XML Spec.
- Namespaces as Map instead of array

Maybe this change should be done after the namespaces refactoring as it would be a breaking change?

Reply:
I'll change this to use the new namespace representation.

If we filter default namespace before, then we have to loop over the list of namespaces twice, but if you are fine with that I can change this.

Reply:
I do not understand what you mean by filter default namespace.
See the key used for default namespace: https://github.com/sap/xml-tools/commit/f9f6fdc#diff-e58b8cf0909522d59cd0ebd69e9fd59aR49